subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/impldeps/ResolvedGeneratedJarsIntegrationTest.groovy

```diff
@@ -29,35 +32,63 @@ class ResolvedGeneratedJarsIntegrationTest extends BaseGradleImplDepsIntegration
         setup:
         productionCode()
 
+        def version = distribution.version.version
+        def generatedJarsDirectory = "user-home/caches/$version/generated-gradle-jars"
+
         when:
         succeeds("tasks")
 
         then:
-        file("user-home/caches/${distribution.version.version}/generated-gradle-jars").assertIsEmptyDir()
+        file(generatedJarsDirectory).assertIsEmptyDir()
 
         when:
         succeeds("classes")
 
         then:
-        file("user-home/caches/${distribution.version.version}/generated-gradle-jars/gradle-api-${distribution.version.version}.jar").assertExists()
+        file("$generatedJarsDirectory/gradle-api-${version}.jar").assertExists()
 
     }
 
     def "gradle testkit jar is generated only when requested"() {
         setup:
         testCode()
 
+        def version = distribution.version.version
+        def generatedJarsDirectory = "user-home/caches/$version/generated-gradle-jars"
+
         when:
         succeeds("classes")
 
         then:
-        file("user-home/caches/${distribution.version.version}/generated-gradle-jars").assertIsEmptyDir()
+        file(generatedJarsDirectory).assertIsEmptyDir()
 
         when:
         succeeds("testClasses")
 
         then:
-        file("user-home/caches/${distribution.version.version}/generated-gradle-jars/gradle-test-kit-${distribution.version.version}.jar").assertExists()
+        file("$generatedJarsDirectory/gradle-test-kit-${version}.jar").assertExists()
+    }
+
+    @Issue(['https://github.com/gradle/gradle/issues/9990', 'https://github.com/gradle/gradle/issues/10038'])
+    def "generated jars (api & test-kit) are valid archives"() {
+        setup:
+        productionCode()
+        testCode()
+
+        def version = distribution.version.version
+        def generatedJarsDirectory = "user-home/caches/$version/generated-gradle-jars"
+        def generatedJars = ["$generatedJarsDirectory/gradle-api-${version}.jar", "$generatedJarsDirectory/gradle-test-kit-${version}.jar"]
```

Comment:
I think this is easier to read and maintain:

```suggestion
        def generatedJars = [
            "$generatedJarsDirectory/gradle-api-${version}.jar",
            "$generatedJarsDirectory/gradle-test-kit-${version}.jar".
        ]
```

Reply:
Committed separately (because of a trailing dot here)

Reply:
👍 (meant to add a trailing comma...)